app/controllers/contestants_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ContestantsController < ApplicationController
+  def index
+    @contestants = Contestant.all
+  end
+
+  def add
+    project = Project.find(params[:project_id])
+    if project.contestants.include?(Contestant.find(params[:contestant_id]))
```

Comment:
Could job handling the edge case when a contestant is already on a project. Ideally, I would also want to see a test for this as well, but understandable that you don't have one since this is a timed challenge. Make sure you are doing it in the project!